packages/accounts-base/accounts_server.js

```diff
@@ -1809,30 +1809,6 @@ function defaultValidateNewUserHook(user) {
 }
 
 const setupUsersCollection = async users => {
-  ///
-  /// RESTRICTING WRITES TO USER OBJECTS
-  ///
-  users.allow({
-    // clients can modify the profile field of their own document, and
-    // nothing else.
-    update: (userId, user, fields, modifier) => {
-      // make sure it is our record
-      if (user._id !== userId) {
-        return false;
-      }
-
-      // user can only modify the 'profile' field. sets to multiple
-      // sub-keys (eg profile.foo and profile.bar) are merged into entry
-      // in the fields list.
-      if (fields.length !== 1 || fields[0] !== 'profile') {
-        return false;
-      }
-
-      return true;
-    },
```

Comment:
> I decided to try out running .update operation on the users collection from the client side and it worked!

Was the .update operation applied to your own user?

Under the allow rule, you can update only your own user, not others, which seems fine, does it?

Reply:
Yes, it was my own record but still. I beg to differ, such case shouldn't be the default way of leaving it open. It should be triggered by the server-side method and if the developer wishes to allow the users he needs to do it explicitly not forcibly shipping it into every Meteor project.

Reply:
>Set default controls on what a user can do against the DB, maybe they can’t change anything or can only do so via specific methods.

It's not about setting 'special' defaults really, it's rather about the absence of it. There shouldn't be any rules that allow custom/special handling of user data beyond what the developer chooses and sets.

 >It’s fine, but it might help to add a test that verifies users can’t modify their data even when we rely on a default deny. Do we have such a test?
 
 Ehh, I'm a bit on edge regarding this. Why should we test something that doesn't exist?! If it's about later on, well when we do add another custom deny/allow rules we should add it.

Reply:
> Ehh, I'm a bit on edge regarding this. Why should we test something that doesn't exist?

Can't you use update methods with User collections? Any collection supports that, right? It does exist.

My point is that if we don’t expect users to ever modify their own data or others, we should handle it. From a design and security perspective, a user collection, the core of any Meteor app, should never allow users to update its or others data. Just restore the test you removed to assert on not allowed modifications.

I don’t have a strong opinion on this, I just think it’s good to have. If anything changes at the allow/deny abstraction level, this test will warn us. And as expected, no one should be able to modify their own or another user’s data.